lib/model/channel.dart

```diff
@@ -398,13 +398,15 @@ class ChannelStoreImpl extends HasUserStore with ChannelStore {
         // details will come in a later `subscription` event.)
 
       case ChannelDeleteEvent():
-        for (final stream in event.streams) {
-          assert(identical(streams[stream.streamId], streamsByName[stream.name]));
-          assert(subscriptions[stream.streamId] == null
-            || identical(subscriptions[stream.streamId], streams[stream.streamId]));
-          streams.remove(stream.streamId);
-          streamsByName.remove(stream.name);
-          subscriptions.remove(stream.streamId);
+        for (final channelId in event.channelIds) {
+          final channel = streams.remove(channelId);
+          if (channel == null) break;
```

Comment:
Hmm, this condition shouldn't happen, right? It'd mean our state was out of sync with the server.

But if it *does* happen, it's best to carry on with removing any other channels that were in the list, so we don't get out of sync in a new way.

So two changes:
```suggestion
        for (final channelId in event.channelIds) {
          final channel = streams.remove(channelId);
          if (channel == null) continue; // TODO(log)
```

The `TODO(log)` comment is a marker for when we someday sweep through and add error reporting through Sentry or something like it, so that we can learn about unexpected conditions the client is hitting in the wild.